src/main/java/uk/gov/hmcts/darts/event/service/handler/DarStartHandler.java

```diff
@@ -1,21 +1,29 @@
 package uk.gov.hmcts.darts.event.service.handler;
 
-import lombok.RequiredArgsConstructor;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 import uk.gov.hmcts.darts.common.entity.EventHandlerEntity;
+import uk.gov.hmcts.darts.common.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.repository.EventRepository;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.service.RetrieveCoreObjectService;
 import uk.gov.hmcts.darts.event.model.DarNotifyApplicationEvent;
 import uk.gov.hmcts.darts.event.model.DartsEvent;
 import uk.gov.hmcts.darts.event.service.handler.base.EventHandlerBase;
 
 import static uk.gov.hmcts.darts.event.enums.DarNotifyType.START_RECORDING;
 
 @Service
-@RequiredArgsConstructor
 public class DarStartHandler extends EventHandlerBase {
 
-    private final ApplicationEventPublisher eventPublisher;
+    public DarStartHandler(RetrieveCoreObjectService retrieveCoreObjectService,
+                           EventRepository eventRepository,
+                           HearingRepository hearingRepository,
+                           CaseRepository caseRepository,
+                           ApplicationEventPublisher eventPublisher) {
+        super(retrieveCoreObjectService, eventRepository, hearingRepository, caseRepository, eventPublisher);
+    }
```

Comment:
author note:
Due to a technical limitation, Lombok is not able to generate constructors that can call non-default superclass constructors. 
Therefore if we want to migrate `EventHandlerBase` to constructor-based injection, we must write subclasses constructors manually. Boilerplate unfortunately.

Either we accept field-based injection in `EventHandlerBase`, or added boilerplate. Please let me know your thoughts.